src/session_manager.rs

```diff
@@ -729,6 +729,37 @@ impl SessionManager {
         }
     }
 
+    pub(crate) fn handle_telegram_link(&self, link: String) {
+        let client_id = self.active_logged_in_client_id();
+        if let Some(client_id) = client_id {
+            spawn(clone!(@weak self as obj => async move {
+                let result = functions::get_internal_link_type(link, client_id).await;
+                if let Ok(link_type) = result {
+                    use enums::InternalLinkType::*;
+                    // TODO: Support other link types
+                    #[allow(clippy::single_match)]
```

Comment:
No need to do that, just follow what clippy says. I know that we'll need a match arm in the future, but I prefer not ignore clippy :)